src/intrinsics/ecma262/BooleanPrototype.js

```diff
@@ -12,20 +12,23 @@
 import type { Realm } from "../../realm.js";
 import { ObjectValue, StringValue, AbstractValue, BooleanValue } from "../../values/index.js";
 import { To } from "../../singletons.js";
-import buildExpressionTemplate from "../../utils/builder.js";
+import { createOperationDescriptor } from "../../utils/generator.js";
 
 export default function(realm: Realm, obj: ObjectValue): void {
   // ECMA262 19.3.1
   obj.$BooleanData = realm.intrinsics.false;
 
-  const tsTemplateSrc = "(A).toString()";
-  const tsTemplate = buildExpressionTemplate(tsTemplateSrc);
-
   // ECMA262 19.3.3.3
   obj.defineNativeMethod("toString", 0, context => {
     const target = context instanceof ObjectValue ? context.$BooleanData : context;
     if (target instanceof AbstractValue && target.getType() === BooleanValue) {
-      return AbstractValue.createFromTemplate(realm, tsTemplate, StringValue, [target], tsTemplateSrc);
+      return AbstractValue.createTemporalFromBuildFunction(
```

Comment:
This makes no sense to me. We know that target is a primitive **boolean** (not undefined and not null) and we know that this version of toString has no side effects. Calling it via the time line is just not right.

There is a bit of an issue with the existing code: All we really know at this point is that there is an expression that ends up calling this toString method on the original value of String.prototype.toString. We don't actually know that (A).toString() will end up calling this method at runtime. So, to be on the safe side we should guard this case with the additional constraint that the current value of String.prototype.toString is in fact this method.

Reply:
I was probably a bit eager here and didn't need to change this one.

Reply:
None of the String.prototype.methods need to become temporal.